src/Compilers/Core/Portable/SourceGeneration/GeneratorDriverCache.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal class GeneratorDriverCache
+    {
+        internal const int MaxCacheSize = 10;
+
+        private readonly (string cacheKey, GeneratorDriver driver)[] _cachedDrivers = new (string, GeneratorDriver)[MaxCacheSize];
```

Comment:
>new (string, GeneratorDriver)[MaxCacheSize] [](http://example.com/codeflow?start=86&length=43)

Credits to @alrz for getting the parsing right here, because my brain assumed this was a target-typed new on first look 😆